app/auth/controllers.py

```diff
@@ -122,54 +115,40 @@ def get_jwt():
             description: Secret key do not match
 
     """
-    try:
-        data = request.get_json()
-        user_name = data.get('username', None)
-        email = data.get('email', None)
-        secret = data.get('secret', None)
-        verify = False
-        user_id = None
-        if user_name is None and email is None:
-            return handle_error('INVALID_INPUT',
-                                'User name or email both can not be empty',
-                                400)
-
-        if secret is None:
-            return handle_error('INVALID_INPUT',
-                                'secret can not be empty',
-                                400)
-        elif user_name is not None:
-            try:
-                user = User.query.filter_by(name=user_name).one()
-            except NoResultFound as e:
-                app.logger.error(e)
-                return handle_error('USER_NOT_FOUND',
-                                    'user does not exists',
-                                    404)
-            if secret == user.secret:
-                verify = True
-                user_id = user.id
-        elif email is not None:
-            try:
-                user = User.query.filter_by(email=email).one()
-            except NoResultFound as e:
-                app.logger.error(e)
-                return handle_error('USER_NOT_FOUND',
-                                    'user does not exists',
-                                    404)
-            if secret == user.secret:
-                verify = True
-                user_id = user.id
-        if verify:
-            return jsonify({'token': JWT(app.config['JWT_SEED'],
-                                         user_id).encode()}), 200
-        else:
-            return handle_error('SECRET_ERROR',
-                                'Secret key do not match',
-                                403)
-    except Exception as e:
-        app.logger.error(e)
-        return handle_error('GENERIC_ERROR', e.message, 500)
+    data = request.get_json()
+    user_name = data.get('username', None)
+    email = data.get('email', None)
+    secret = data.get('secret', None)
+    verify = False
+    user_id = None
+    if user_name is None and email is None:
+        raise InvalidUsage('User name or email both can not be empty', 400)
```

Comment:
I think it is "User name and email cannot both be empty". Note I'm not even clear where this error message comes from - when do we submit both of these?